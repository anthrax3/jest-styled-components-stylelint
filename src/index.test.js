require('./index')({ failOnError: false })
require('jest-styled-components')
const React = require('react')
const { default: styled, injectGlobal } = require('styled-components')
const TestRenderer = require('react-test-renderer')

test('causes styles with lint to log errors', () => {
  // prettier-ignore
  injectGlobal`
    .test {
      test-decoration: underlin
    }
  `
  // prettier-ignore
  const Title = styled.h1`
    color: red;
    line-height: -
  `
  const wrapper = TestRenderer.create(
    <Title>Who wrote these styles anyway??</Title>
  )
  expect(wrapper.toJSON()).toMatchSnapshot()
})

test('styles without lint log nothing', () => {
  injectGlobal`
    .test {
      text-decoration: underline;
    }
  `
  // prettier-ignore
  const Title = styled.h1`
    color: red;
  `
  TestRenderer.create(<Title />)
})

test('does not pass stylelint', () => {
  // prettier-ignore
  const Title = styled.h1`
    color: {oops!}
  `
  TestRenderer.create(<Title />)
})

test('works on nested style rules', () => {
  // prettier-ignore
  const Link = styled.a`
    text-decoration: undrln
  `
  // prettier-ignore
  const Title = styled.h1.attrs({
    children: <Link />
  })`
    color: red;

    a:link {
      text-decoration: 5;


      @media (min-width: ) {
        width: 10frobs
      }
    }
  `
  TestRenderer.create(<Title />)
})

test('handles styles without newlines', () => {
  // prettier-ignore
  injectGlobal`.selector { color:black }`
  // prettier-ignore
  const Link = styled.a`text-decroation: underline`
  TestRenderer.create(<Link />)
})

test('handles empty styles', () => {
  // prettier-ignore
  injectGlobal``
  // prettier-ignore
  const Link = styled.a``
  TestRenderer.create(<Link />)
})

test('handles styles with weird newline situations', () => {
  // prettier-ignore
  const Link = styled.a`text-decroation: value(
      one,
      two,
      three)
  `
  TestRenderer.create(<Link />)
})
