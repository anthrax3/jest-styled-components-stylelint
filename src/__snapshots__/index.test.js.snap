// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`causes styles with lint to log errors 1`] = `
.c0 {
  color: red;
  line-height: -;
}

<h1
  className="c0"
>
  Who wrote these styles anyway??
</h1>
`;
